internal/test/framework/windowsvm.go

```diff
@@ -1,98 +1,82 @@
-	Run(string, bool) (string, string, error)
-	// Run executes the given command remotely on the Windows VM over a ssh connection and returns the combined output
-	// of stdout and stderr. If the bool is set, it implies that the cmd is to be execute in PowerShell. This function
-	// should be used in scenarios where you want to execute a command that runs in the background. In these cases we
-	// have observed that Run() returns before the command completes and as a result killing the process.
-	RunOverSSH(string, bool) (string, error)
-	// GetCredentials returns the interface for accessing the VM credentials. It is up to the caller to check if non-nil
-	// Credentials are returned before usage.
-	GetCredentials() *types.Credentials
-	// Reinitialize re-initializes the Windows VM. Presently only the ssh client is reinitialized.
-	Reinitialize() error
 	// Destroy destroys the Windows VM
+	// TODO: Remove this and move it to framework or other higher level object capable of doing deletions.
+	//		jira: https://issues.redhat.com/browse/WINC-243
 	Destroy() error
+	// Embed the test
+	types.WindowsVM
 }
 
 // newWindowsVM creates and sets up a Windows VM in the cloud and returns the WindowsVM interface that can be used to
 // interact with the VM. If credentials are passed then it is assumed that VM already exists in the cloud and those
 // credentials will be used to interact with the VM. If no error is returned then it is guaranteed that the VM was
 // created and can be interacted with. If skipSetup is true, then configuration steps are skipped.
-func newWindowsVM(imageID, instanceType string, credentials *types.Credentials, skipSetup bool) (WindowsVM, error) {
-	w := &windowsVM{}
+func newWindowsVM(imageID, instanceType string, credentials *types.Credentials, skipSetup bool) (TestWindowsVM, error) {
+	w := &testWindowsVM{}
 	var err error
 
-	w.cloudProvider, err = cloudprovider.CloudProviderFactory(kubeconfig, awsCredentials, "default", artifactDir,
+	cloudProvider, err = cloudprovider.CloudProviderFactory(kubeconfig, awsCredentials, "default", artifactDir,
 		imageID, instanceType, sshKey, privateKeyPath)
 	if err != nil {
 		return nil, fmt.Errorf("error instantiating cloud provider %v", err)
 	}
 
 	if credentials == nil {
-		w.credentials, err = w.cloudProvider.CreateWindowsVM()
+		windowsVM, err := cloudProvider.CreateWindowsVM()
 		if err != nil {
 			return nil, fmt.Errorf("error creating Windows VM: %v", err)
 		}
+		// TypeAssert to the WindowsVM struct we want
+		winVM, ok := windowsVM.(*types.Windows)
+		if !ok {
+			return nil, fmt.Errorf("error asserting Windows VM to %v", err)
+		}
+		w.Windows = winVM
 	} else {
+		//TODO: Add username as well, as it will change depending on cloud provider
 		if credentials.GetIPAddress() == "" || credentials.GetPassword() == "" {
 			return nil, fmt.Errorf("password or IP address not specified in credentials")
 		}
-		w.credentials = credentials
+		w.Windows.Credentials = credentials
```

Comment:
Won't referencing `w.Credentials` works and make this more succinct? This comment applies to all instances if this is possible.